lib/windshaft/renderers/blend/factory.js

```
@@ -1,123 +1,114 @@
+                    onTileErrorStrategyHttpRenderer :
+                    onTileErrorStrategy;
 
-            var rendererOptions = {
-                limits: limits
-            };
+                if (isMapnikLayer) {
+                    hasMapnikLayer = true;
+                }
 
-            // Avoid failing because of timeouts or 404/errors in http layers
-            if (layerType === 'http') {
-                rendererOptions.onTileErrorStrategy = function(err, tile, headers, stats, format, callback) {
-                    if (err.code === 'ETIMEDOUT') {
-                        return callback(err, tile, headers, stats);
-                    }
-                    return callback(null, EMPTY_IMAGE_BUFFER, {'Content-Type': 'image/png'}, stats || {});
-                };
-            } else {
-                rendererOptions.onTileErrorStrategy = function(err, tile, headers, stats, format, callback) {
-                    if (err.message && err.message.match(/coordinates out of range/i)) {
-                        return callback(null, EMPTY_IMAGE_BUFFER, {'Content-Type': 'image/png'}, stats || {});
+                this.rendererFactory.getRenderer(mapConfig, rendererParams, rendererOptions, (err, renderer) => {
+                    if (err) {
+                        return reject(err);
                     }
-                    return callback(err, tile, headers, stats);
-                };
-            }
 
-            self.rendererFactory.getRenderer(mapConfig, rendererParams, rendererOptions, cb);
+                    return resolve(renderer);
+                });
+            });
+        });
 
-            if (isMapnikLayer) {
-                hasMapnikLayer = true;
-            }
-        }, params, mapConfig);
-    });
+        return Promise.all(layerRenderers)
+            .then(renderers => {
+                if (!Array.isArray(renderers) || !renderers.length) {
+                    throw new Error('No renderers');
```

Why are we throwing here instead of `callback(new Error('No renderers'));`?

---

We can use both, inside a `then-able` you can throw an error and it will be handled in the next `.catch` of the chain. I'm throwing the error just for the shake to have on a place to handle errors in that promises' chain. Imagine you need to differentiate errors for some reason:

```js
return Promise.all(layerRenderers)
            .then(renderers => {
                if (!Array.isArray(renderers) || !renderers.length) {
                    throw new Error('No renderers');
                }
                [ ... ]
             })
             .catch(err => err instanceof WadusError ? callback() : callback(err));
```